internal/pkg/v2/infrastructure/redis/event.go

```diff
@@ -84,6 +83,34 @@ func addEvent(conn redis.Conn, e models.Event) (addedEvent models.Event, edgeXer
 	return e, edgeXerr
 }
 
+func deleteEvent(conn redis.Conn, id string) (edgeXerr errors.EdgeX) {
+	// query Event by Id first to ensure there is an corresponding event
+	e, edgeXerr := eventById(conn, id)
+	if edgeXerr != nil {
+		return edgeXerr
+	}
+
+	storedKey := fmt.Sprintf("%s:%s", EventsCollection, id)
+	_ = conn.Send(MULTI)
+	_ = conn.Send(UNLINK, storedKey)
+	_ = conn.Send(UNLINK, EventsCollection+":readings:"+storedKey)
+	_ = conn.Send(ZREM, EventsCollection, storedKey)
+	_ = conn.Send(ZREM, EventsCollection+":created", storedKey)
+	_ = conn.Send(ZREM, EventsCollection+":pushed", storedKey)
+	_ = conn.Send(ZREM, EventsCollection+":device:"+e.DeviceName, storedKey)
+
+	res, err := redis.Values(conn.Do(EXEC))
```

Comment:
Does this also delete the events readings? Or should be calling `deleteReading()`?

Reply:
Ooops, I shall invoke `deleteReadingById` as well.  Thanks for the catch!

Reply:
since aside from pushed, event and reading are both immutable, we don't need to recreate readings.
it consumes lots of computing resource to update readings for just a pushed if there are more than 100 readings in an event.

Reply:
After further consideration, I've revised the codes to only update database records that are associated with `pushed` timestamp.